pallets/nfts/src/features/approvals.rs

```diff
@@ -165,10 +178,255 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 		Self::deposit_event(Event::AllApprovalsCancelled {
 			collection,
-			item,
+			item: Some(item),
 			owner: details.owner,
 		});
 
 		Ok(())
 	}
+
+	/// Approves the transfer of all items within the collection to a delegate.
+	///
+	/// This function is used to approve the transfer of all items within the `collection` to
+	/// a `delegate`. If `maybe_check_origin` is specified, the function ensures that the
+	/// `check_origin` account is the owner of the collection, granting them permission to approve
+	/// the transfer. The `delegate` is the account that will be allowed to take control of all
+	/// items within the collection. Optionally, a `deadline` can be specified to set a time limit
+	/// for the approval. The `deadline` is expressed in block numbers and is added to the current
+	/// block number to determine the absolute deadline for the approval. After approving the
+	/// transfer, the function emits the `TransferApproved` event.
+	///
+	/// - `maybe_check_origin`: The optional account that is required to be the owner of the item,
+	///   granting permission to approve the transfer. If `None`, no permission check is performed.
+	/// - `collection`: The identifier of the collection.
+	/// - `delegate`: The account that will be allowed to take control of all items within the
+	///   collection.
+	/// - `maybe_deadline`: The optional deadline (in block numbers) specifying the time limit for
+	///   the approval.
+	pub(crate) fn do_approve_collection(
+		maybe_check_origin: Option<T::AccountId>,
+		collection: T::CollectionId,
+		delegate: T::AccountId,
+		maybe_deadline: Option<frame_system::pallet_prelude::BlockNumberFor<T>>,
+	) -> DispatchResult {
+		ensure!(
+			Self::is_pallet_feature_enabled(PalletFeature::Approvals),
+			Error::<T, I>::MethodDisabled
+		);
+
+		let collection_config = Self::get_collection_config(&collection)?;
+		ensure!(
+			collection_config.is_setting_enabled(CollectionSetting::TransferableItems),
+			Error::<T, I>::ItemsNonTransferable
+		);
+
+		let (owner, deadline) = Collection::<T, I>::try_mutate(
+			collection,
+			|maybe_collection_details| -> Result<(T::AccountId, Option<BlockNumberFor<T>>), DispatchError> {
+				let collection_details =
+					maybe_collection_details.as_mut().ok_or(Error::<T, I>::UnknownCollection)?;
+				let owner = collection_details.clone().owner;
+
+				if let Some(check_origin) = maybe_check_origin {
+					ensure!(check_origin == owner, Error::<T, I>::NoPermission);
+				}
+				let now = frame_system::Pallet::<T>::block_number();
+				let deadline = maybe_deadline.map(|d| d.saturating_add(now));
+
+				AccountApprovals::<T, I>::try_mutate(
+					collection,
+					&owner,
+					|allowances| -> Result<(), DispatchError> {
+						ensure!(
+							*allowances < T::ApprovalsLimit::get(),
```

Comment:
Explanation: Makes sure the number of collection approvals is under the configured limit